src/CargoBuild/CargoBuildTask.cs

```diff
@@ -0,0 +1,437 @@
+        private static readonly string _msRustupBinary = $"{_tempPath}\\cargohome\\bin\\msrustup.exe";
+        private static readonly Dictionary<string, string> _envVars = new () { { "CARGO_HOME", _cargoHome }, { "RUSTUP_HOME", _rustupHome }, };
+        private static readonly string _rustupDownloadLink = "https://static.rust-lang.org/rustup/dist/x86_64-pc-windows-msvc/rustup-init.exe";
+        private static readonly string _checkSumVerifyUrl = "https://static.rust-lang.org/rustup/dist/x86_64-pc-windows-msvc/rustup-init.exe.sha256";
+        private bool _shouldCleanRustPath;
+        private string? _currentRustupInitExeCheckSum;
+
+        private enum ExitCode
+        {
+            Succeeded,
+            Failed,
+        }
+
+        /// <summary>
+        /// Gets or sets a cargo command to execute.
+        /// </summary>
+        [Required]
+        public string Command { get; set; } = string.Empty;
+
+        /// <summary>
+        /// Gets or sets start up project path.
+        /// </summary>
+        [Required]
+        public string StartupProj { get; set; } = string.Empty;
+
+        /// <summary>
+        /// Gets or sets a value indicating whether to perform authorization.
+        /// </summary>
+        public bool EnableAuth { get; set; } = false;
+
+        /// <summary>
+        /// Gets or sets a feed name.
+        /// </summary>
+        public string RegistryFeedName { get; set; } = string.Empty;
+
+        /// <summary>
+        /// Gets or sets optional cargo command args.
+        /// </summary>
+        public string CommandArgs { get; set; } = string.Empty;
+
+        /// <summary>
+        /// Gets or sets a value indicating whether to use msrust up or not.
+        /// </summary>
+        public bool UseMsRustUp { get; set; } = false;
+
+        /// <inheritdoc/>
+        public override bool Execute()
+        {
+            // download & install rust if necessary
+            if (DownloadRustupAsync().GetAwaiter().GetResult())
+            {
+                if (InstallRust().GetAwaiter().GetResult())
+                {
+                    _shouldCleanRustPath = true;
+                }
+            }
+
+            if (Command.Equals("clearcargocache", StringComparison.InvariantCultureIgnoreCase))
+            {
+                if (Directory.Exists(_cargoHome))
+                {
+                    Log.LogMessage(MessageImportance.Normal, $"Clearing cargo cache at {_cargoHome}");
+                    Directory.Delete(_cargoHome, true);
+                }
+            }
+
+            if (!Command.Equals("fetch", StringComparison.InvariantCultureIgnoreCase))
+            {
+                var dir = Directory.GetParent(StartupProj!);
+                bool cargoFile = File.Exists(Path.Combine(dir!.FullName, "cargo.toml"));
+
+                return !cargoFile || CargoRunCommandAsync(Command.ToLower(), CommandArgs).GetAwaiter().GetResult() == ExitCode.Succeeded;
```

Comment:
This should log an explicit error when the cargo file doesn't exist. What you have now will just fail the task with no error for the user.

Reply:
Actually, if the file doesn't exist, it returns `true` with no error? This doesn't seem correct.